### ISSUE TITLE
Optimizations for "multi-site" use case

### ISSUE THREAD

**Issue by aryascripts** [magidoc-org/magidoc#199]
## The Problem
Currently, there seems to be no way to differentiate between "static" content, and "_app" content. This means that the svelte app / JS portions are generated statically, and we have to generate both, any time either one changes. This may be okay for single site use cases, but if this library is used for multiple sites, then this can really take a toll on the static site-building and cost money to run in CI.

## Example

If today, we start with the sidebar / external links of:
- Site 1
- Site 2
- Site 3

And now, we want to add:
- Site 4

Then:
All 4 sites (the new one, and 3 older ones) need to be rebuilt. Even though the only change needed for Site 1, 2, and 3 is to update "External Links".

### Describe the proposed solution

Have a process where the following elements can be generated and replaced without having to rebuild all sites:
- sidebar external links
- logo / site names (maybe?)

### Importance

Nice to have

**Comment by amanb014** [comment]
Yeah that totally makes sense. I don't really see another way around this except for building sidebar separately and somehow connecting the two together. Though the performance of generating sites is not huge, unless the graphql schema has 100s of queries/mutations to generate. Most sites generate in under 1 minute so it might not be totally out of the ordinary to just generate all of them again. 

For now, for my use case I will probably go with generating sites individually without any external links. And the "linking" between sites will happen outside of Magidoc completely and not hooked in with the site generation. That may be a better outcome for me anyways. Thanks!
